### PR TITLE
Migrate to JUnit5

### LINKED ISSUE #408: Bump up JUnit version to 5
Presently JUnit 5 is under alpha stage, once it is open for public use we should update library. 

JUnit 5 is lambda ready and provides variety of features pre-built that will remove some library dependencies.

Plus it never hurts to stay updated and we can provide feedback on any broken features.

### PR DESCRIPTION
**Pull request by amogozov**
Migrate project to JUnit5 
- Switched all modules to use junit platform
- If it is impossible to convert module to junit 5 engine, use vintage engine instead (run tests as junit4)
- Should potentially resolve #408

### CODE REVIEW

File: dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java [hunk 1]
```diff
@@ -184,20 +181,21 @@ public void updationShouldBeSuccessAndAccessingTheSameCustomerShouldReturnUpdate
    * DB service unavailable.
    * 
    */
+  @Nested
   public class ConnectivityIssue {
     
     private static final String EXCEPTION_CAUSE = "Connection not available";
-    @Rule public ExpectedException exception = ExpectedException.none();
+    //@Rule public ExpectedException exception = ExpectedException.none();
```
Comment (iluwatar) [inline dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java]: Remove commented code

File: dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java [hunk 2]
```diff
@@ -184,20 +181,21 @@ public void updationShouldBeSuccessAndAccessingTheSameCustomerShouldReturnUpdate
...
      * @throws SQLException if any error occurs.
      */
-    @Before
+    @BeforeEach
     public void setUp() throws SQLException {
       dao = new DbCustomerDao(mockedDatasource());
-      exception.expect(Exception.class);
-      exception.expectMessage(EXCEPTION_CAUSE);
+      //exception.expect(Exception.class);
+     //exception.expectMessage(EXCEPTION_CAUSE);
```
Comment (iluwatar) [inline dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java]: Remove commented code

**Comment by iluwatar** [comment]
Well done @amogozov Many thanks for this great PR. :+1:

File: dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java (hunk 2 above)
Comment (amogozov) [reply dao/src/test/java/com/iluwatar/dao/DbCustomerDaoTest.java]: Sorry for this leftover, let me submit separate PR

**Comment by Quantriple** [comment]
Exception in thread "main" java.lang.NoSuchMethodError: org.junit.platform.commons.util.ReflectionUtils.getDefaultClassLoader()Ljava/lang/ClassLoader

I've looked at a lot of ways to change the juit version but I still have an error and ask for help

**Pull request merged**